sdk/src/azure/core/az_http_policy_retry.c

```diff
@@ -75,12 +75,12 @@ AZ_INLINE void _az_http_policy_retry_log(int32_t attempt, int32_t delay_msec)
 AZ_INLINE AZ_NODISCARD int32_t _az_uint32_span_to_int32(az_span span)
 {
   uint32_t value = 0;
-  if (az_result_succeeded(az_span_atou32(span, &value)))
+  if (az_result_failed(az_span_atou32(span, &value)))
```

Comment:
Checking for failure first since that is the exceptional code path.

Reply:
nice